### ISSUE TITLE
Phpredis 6.0 psubscribe() with wildcards will instantly stop after receive first message

### ISSUE THREAD

**Issue by asika32764** [phpredis/phpredis#2395]
### Test Code

```php
$redis = new Redis();
$redis->connect('127.0.0.1');

$r = $redis->psubscribe(
    ['foo.*'],
    function () {
        var_dump('Received');
    }
);

var_dump('Subscribe STOP', $r);
```

And run this command:

```
redis-cli PUBLISH "foo.1" "Hello"
```

### Expected behaviour

Receive message and show `Received`

### Actual behaviour

Stop subscribe and show:

```
string(14) "Subscribe STOP"
bool(false)
```

### Other Test cases

- Use `subscribe()` can work.
- Use `psubscribe()` without wildcards also works.
- Only `psubscribe()` with wildcard pattern will break.

### Screen Recording

6.0.1 not work

![2023-09-25 20 20 06](https://github.com/phpredis/phpredis/assets/1639206/ea4a2479-8641-46c9-8829-a63e9c4374da)

But 5.3.7 works

![2023-09-25 20 21 02](https://github.com/phpredis/phpredis/assets/1639206/8b826d40-db3f-4056-b695-fc89f83d254d)

### I'm seeing this behaviour on
- OS: MacOS / Ubuntu
- Redis: 6.0.8
- PHP: 8.2
- phpredis: 6.0.1

### I've checked
- [x] There is no similar issue from other users
- [ ] Issue isn't fixed in `develop` branch

**Comment by yatsukhnenko** [comment]
It looks like error occurred when calling your callback function

**Comment by asika32764** [comment]
Yes but this is a simple callback without any logic.

**Comment by michael-grunder** [comment]
I can replicate this.  What's happening is that PhpRedis 6.0.0 is searching for `foo.1` in the callback lookup but should be searching for the pattern, `foo.*`.

Fix should be simple

**Comment by michael-grunder** [comment]
Fixed in #2396
